### ISSUE TITLE
PhantomJS 2.x support?

### ISSUE THREAD

**Issue by dlong500** [pofider/phantom-html-to-pdf#39]
Is there anything blocking this module (or others like phantom-workers) from having the dependency form PhantomJS updated to the 2.x series?

**Comment by pofider** [comment]
I'm carefully monitoring phantomjs forums and issues and there are still many problems with phantomjs 2.x and I think that phantomjs 1.9 is better choice for pdf rendering by now.

Main issues
1. scaling issue - it produces wrong sized font sizes
2. no support for thead
3. future releases will remove pdf headers functionality
4. the maintenance team is understaffed, the last release is from January

I think that the best I can do now is remove the default phantomjs and let the lib user to install and provide the path to phantomjs manually.  What you think?

**Comment by dlong500** [comment]
I think that's a good idea.  I was going to recommend making it one of the `peerDependencies` and allowing for either v1 or v2 of the phantomjs module, but it's even more complicated because the module changed names after hitting 2.x (now it's called `phantomjs-prebuilt`).

So allowing the user to manually pull in whatever version of phantomjs they want sounds like a good idea (as long as you still provide information on a version you've tested and recommended).  The issue for me is more of an annoyance just trying to keep NPM from constantly complaining about old packages (since the version of phantomjs currently required still uses the deprecated npmconf module).

**Comment by dlong500** [comment]
I just checked, and it looks like some of the older dependencies like npmconf were removed in the latest 1.x version of the `phantomjs` module (1.9.20).  So could we at least bump the dependency up to that version for `phantom-html-to-pdf`  and `phantom-workers` even if the longer term solution is to let the user handle things separately?

**Comment by jpike88** [comment]
What's the status of this issue?

**Comment by codecvlt** [comment]
Yea, there are currently security issues with phantomjs 1.9x. It would be really great, if the default could be updated to use the 2.x version.

Thanks!

**Comment by bjrmatos** [comment]
regarding the security issues with phantomjs 1.9.x i've made a comment [here](https://github.com/pofider/phantom-html-to-pdf/issues/69)

**Comment by dlong500** [comment]
Based on the lack of maintenance over the past couple years and [this post](https://groups.google.com/forum/#!topic/phantomjs/9aI5d-LDuNE) it looks to me like PhantomJS is going away at some point (or at least is going to be stagnant).  It looks like switching to an alternative (possibly the new headless Chrome mode) would be the best long term solution.  I'm sure that will entail a decent amount of work, but as the rest of the javascript ecosystem evolves things will most likely only get harder/buggier trying to stick with PhantomJS.

**Comment by bjrmatos** [comment]
> Based on the lack of maintenance over the past couple years and this post it looks to me like PhantomJS is going away at some point (or at least is going to be stagnant). It looks like switching to an alternative (possibly the new headless Chrome mode) would be the best long term solution.

yes, exactly. we have plans to make a package that uses Chrome healdess mode with the same use case of this package. 😀

**Comment by taoqf** [comment]
Bad news: this doesn't work under nodejs v12.x
